notebooks/219-colorization/219-colorization.ipynb

```diff
@@ -0,0 +1,288 @@
+    "    img = np.squeeze(img)\n",
+    "    # (C, H, W) -> (H, W, C)\n",
+    "    img = img.transpose((1, 2, 0))\n",
+    "    img = cv.resize(img, (W, H))\n",
+    "    img_lab_full = np.concatenate((img_lab[:, :, 0][:, :, np.newaxis], img), axis=2)\n",
+    "    img_bgr = np.clip(cv.cvtColor(img_lab_full, cv.COLOR_Lab2BGR), 0, 1)\n",
+    "    img_bgr = (img_bgr * 255).astype(np.uint8)\n",
+    "    return img_bgr\n",
+    "\n",
+    "\n",
+    "def show_input_output_image(input_image, output_image):\n",
+    "    \"\"\"\n",
+    "    show input image and output image\n",
+    "    \"\"\"\n",
+    "    fig = plt.figure(figsize=(10, 7))\n",
+    "    rows = 1\n",
+    "    columns = 2\n",
+    "\n",
+    "    fig.add_subplot(rows, columns, 1)\n",
+    "    plt.imshow(input_image)\n",
+    "    plt.axis('off')\n",
+    "    plt.title(\"Input\")\n",
+    "\n",
+    "    fig.add_subplot(rows, columns, 2)\n",
+    "    plt.imshow(output_image)\n",
+    "    plt.axis('off')\n",
+    "    plt.title(\"Output\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run Inference\n",
+    "\n",
+    "To run inference, we need to create infer request. `colorization-siggraph` model takes 3 inputs: `data_l`, `user_ab`, `user_map`, we  don't need to specify all three inputs to use the model. We can set the input `data_l` to the preprocessed image, and the remaining inputs `user_ab`, `user_map` to zeros. We can call request.infer() method to run inference, then get output from result tensors."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_tensor_name = 'data_l'\n",
+    "output_tensor = compiled_model.outputs[0]\n",
+    "\n",
+    "inputs = {}\n",
+    "for input in model.inputs:\n",
+    "    inputs[input.get_any_name()] = np.zeros(input.shape)\n",
+    "inputs.keys()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def colorize(image_path):\n",
+    "    \"\"\"\n",
+    "    image_path can be local path or URL\n",
+    "    \"\"\"\n",
+    "    H, W = 256, 256\n",
+    "    image = load_image(image_path)\n",
+    "    (h, w, c) = image.shape\n",
+    "    img_lab, img_l_channel = preprocess(image, W, H)\n",
+    "    inputs[input_tensor_name] = np.expand_dims(img_l_channel, axis=[0, 1])\n",
+    "\n",
+    "    # create infer request and perform inference\n",
+    "    infer_request = compiled_model.create_infer_request()\n",
+    "    res = infer_request.infer(inputs)[output_tensor]\n",
+    "\n",
+    "    out = postprocess(res, img_lab, w, h)\n",
+    "    show_input_output_image(cv.cvtColor(image, cv.COLOR_BGR2RGB), cv.cvtColor(out, cv.COLOR_BGR2RGB))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "colorize('lena.png')"
```

Comment:
Please use a different image to test the colorize task. Although Lena has been widely used in the computer vision field, the CV community has received strong criticism for its excessive use and made a call to reduce its use.  
https://pursuit.unimelb.edu.au/articles/it-s-time-to-retire-lena-from-computer-science

Try to use URLs for the images you want to use here, following the AI Ethics we have in the Contributing.md guideline.